### PR TITLE
Navbar conditional rendering

### LINKED ISSUE #71: FE: Filter navbar items based on account type
Changes include the following:
- Right now, every item in the navbar will render based off the authentication state. We also want these to conditionally render if they have access to that page. `canView` in Navbar determines this. If the profile's account type is in the array than we can let the user proceed.

### PR DESCRIPTION
**Pull request by Abhi-S-1**
### What this PR does (required):
- Makes it so that certain links in the navbar are viewed only based on your account type.

### Screenshots / Videos (front-end only):
"navbar pet owner"

"navbar pet sitter"

"navbar logged out"

### Any information needed to test this feature (required):
- Must have a profile in mongodb with accountType as either pet_owner or pet_sitter. Switch between both, as well as logging out on the app to see the three different navbar states

### CODE REVIEW

**Comment by Abhi-S-1** [comment]
Let me know if you want anything changed!

File: client/src/components/Navbar/Navbar.tsx [hunk 1]
```diff
@@ -112,10 +113,13 @@ const Navbar: React.FC = () => {
   };
 
   const renderMenuItems = () => {
-    // TODO: conditionally render based on profile type
     return menuItems.map((menu) => {
       if (menu.authenticated) {
-        return loggedInUser && <MenuItem key={menu.resource} {...menu} />;
+        if (profile) {
+          if (!menu.canView || menu.canView == profile.accountType || menu.canView.length > 1) {
```
Comment (moffatethan) [inline client/src/components/Navbar/Navbar.tsx]: Not quite the approach that was expected. If the account type is included in the canView list then render the route.

File: server/models/User.js [hunk 2]
```diff
@@ -2,6 +2,10 @@ const mongoose = require("mongoose");
 const bcrypt = require("bcryptjs");
 
 const userSchema = new mongoose.Schema({
+  accountType: {
+    type: String,
+    default: "pet_owner"
+  },
```
Comment (moffatethan) [inline server/models/User.js]: Now that we have separated let's keep this out

**Review by moffatethan — CHANGES_REQUESTED** [review]

**Review by moffatethan — CHANGES_REQUESTED** [review]
Please address merge conflicts

**Comment by RetiFier** [comment]
> Please address merge conflicts

@moffatethan Merge Conflicts has been resolved

**Review by moffatethan — APPROVED** [review]

**Pull request merged**